### DISCUSSION TITLE
realtek 8168/8169 doesn't work

### DISCUSSION THREAD

**Discussion by skeletorgithub** [ipxe/ipxe#1280] [Q&A]
Hello.
We have a box with motherboard asrock j4125b-itx with embedded NIC, which detected by LiveCD as:
```
02:00.0 Ethernet controller [0200]: Realtek Semiconductor Co., Ltd. RTL8111/8168/8411 PCI Express Gigabit Ethernet Controller [10ec:8168] (rev 15)

driver: r8169
version: 6.1.0-22-amd64
firmware-version: rtl8168h-2_0.0.2 02/26/15
expansion-rom-version: 
bus-info: 0000:02:00.0
supports-statistics: yes
supports-test: no
supports-eeprom-access: no
supports-register-dump: yes
supports-priv-flags: no
```
but it doesn't detect by iPXE (we use 1.21.1):
```
downloading NBP file ok
autoexec.ipxe ok
debug(pci):
(10ec:8168) has driver "rtl8168"
...
probe failed: Connection timed out
```
According to hardware list, this NIC supported by iPXE. Also, I've tried to change in sources to force using driver r8169, but it didn't help.
```
iPXE initialising devices...Adding 3c509 root bus
Adding EFI root bus
0000:02:00.0 (10ec:8168) has driver "rl18169"
0000:02:00.0 has mem a1004000 io e000 irq 255
0000:02:00.0 latency timer is unreasonably low at 0. Setting to 32.
0000:02:00.0 probe failed: Connection timed out (https://ipxe.org/4c634092)
```

Could you help me?

**Comment by mcb30** [comment]
> According to hardware list, this NIC supported by iPXE. Also, I've tried to change in sources to force using driver r8169, but it didn't help.

Don't change the source.  iPXE will already use the `realtek` driver for that PCI ID.  Please remove **all** of your source modifications before doing any further debugging.

> ```
> 0000:02:00.0 latency timer is unreasonably low at 0. Setting to 32.
> 0000:02:00.0 probe failed: Connection timed out (https://ipxe.org/4c634092)
> ```

As per the instructions on that page https://ipxe.org/4c634092, please build with `DEBUG=realtek` to find out what is going wrong.

**Comment by skeletorgithub** [comment]
Thank you for reply.
Here you are:

![signal-2024-08-22-144429_004](https://github.com/user-attachments/assets/0dc3d6bf-bcab-4b11-aa1a-230b0e6a46ba)
![signal-2024-08-22-144429_003](https://github.com/user-attachments/assets/4176eaeb-1b71-48ee-b3c6-51b06f2d7923)
![signal-2024-08-22-144429_002](https://github.com/user-attachments/assets/1cc40055-9c06-4eb0-a8b7-a1503feed92b)

**Reply by mcb30** [reply]
Thank you.  Please remove all of the other debugging options that you have also added, build with only `DEBUG=realtek:15`, and post the output from that.

Thanks,

Michael

**Comment by skeletorgithub** [comment]
```
mem a1004037 => ff 
```
repeats many many times, so, beginning and ending below:
![signal-2024-08-22-154356_003](https://github.com/user-attachments/assets/45b7508a-299b-48d2-9fc4-ce5da8804e57)
![signal-2024-08-22-154356_002](https://github.com/user-attachments/assets/91d8db92-a9e9-4568-ad95-aa88b5d6a40d)
